en/micro-integrator/docs/develop/debugging-on-server.md

```diff
@@ -0,0 +1,80 @@
+# Debugging On Server
+
+Once you [deploy and run](../deploy-artifacts) your integration solution, you may encounter errors and identify the required modifications for your class mediators. Use the debug on server feature in WSO2 Integration Studio to troubleshoot errors.
+
+There are two ways to debug a developed class mediator.
+
+1.  Instant debugging using the embedded Micro Integrator of WSO2 Integration Studio.
+2.  Deploy artifacts to an external Micro Integrator server and debug.
+
+Above two approaches are discussed in detail below.
+
+## Instant debugging using Micro Integrator
+
+1.  When project artifacts are ready, select the project you want to debug and click **Run** -> **Debug As** -> **Debug on Server**.
+    ![debug-on-server](../assets/img/debug-on-server/select-debug-on-server.png)
+
+2.  In the Debug On Server dialog box that opens, select the required server and click **Next**.
+    ![select debugging](../assets/img/debug-on-server/select-server.png)
+
+3.  Add the artifacts that need to be deployed to the embedded Micro Integrator and click **Finish**.
+    ![select debugging](../assets/img/debug-on-server/add-artifacts.png)
+
+4.  On the console of WSO2 Integration Studio, notice that Micro
+    Integrator is started with the artifacts deployed. HTTP traffic is
+    listened on the 8290 port.
+
+5.  Add some breakpoints to the Java class used in the class mediator.  
+    ![add-breakpoint](../assets/img/debug-on-server/add-breakpoints.png)
+
+6.  Invoke the service using the inbuilt HTTP client or some external client. As soon as
+    a request comes to the class mediator, the first break point will be triggered.  
+    ![select debugging](../assets/img/debug-on-server/debug-class-mediator-view-properties.png)
+    Note that you can view the message context that comes into the mediator.
+
+    !!! Note
+        If you get the error "**Source not found**" follow the instructions below to add the class mediator source.
+        
+        1.  Click **Edit Source Lookup Path**.
```

Comment:
where does this option relies ( shall we mention in brackets, if we can't add screen shots in notes )

Reply:
Fixed with d5e0be9